src/scripts/validation.js

```diff
@@ -0,0 +1,78 @@
+function showInputError(form, input, errorMessage, config) {
+  const errorElement = form.querySelector(`#${input.id}-error`);
+
+  errorElement.textContent = errorMessage;
+  errorElement.classList.add(config.errorClass);
+
+  input.classList.add(config.inputErrorClass);
+}
+
+function hideInputError(form, input, config) {
+  const errorElement = form.querySelector(`#${input.id}-error`);
+
+  errorElement.textContent = "";
+  errorElement.classList.remove(config.errorClass);
+
+  input.classList.remove(config.inputErrorClass);
+}
+
+function checkInputValidity(form, input, config) {
+  if (input.validity.valid) {
+    hideInputError(form, input, config);
+  } else {
+    showInputError(form, input, input.validationMessage, config);
+  }
+}
+
+export function checkFormValidity(form, config) {
+  const inputs = Array.from(form.querySelectorAll(config.inputSelector));
+  const button = form.querySelector(config.submitButtonSelector);
+
+  toggleButtonState(inputs, button);
+
+  inputs.forEach((input) => {
+    checkInputValidity(form, input, config);
+    toggleButtonState(inputs, button);
+  });
+}
+
+function toggleButtonState(inputs, button) {
+  const notValid = inputs.some((input) => {
+    return !input.validity.valid;
+  });
+
+  if (notValid) {
+    disableButton(button);
+  } else {
+    enableButton(button);
+  }
+}
+
+function enableButton(button) {
+  button.disabled = false;
+}
+
+function disableButton(button) {
+  button.disabled = true;
+}
+
+function setEventListeners(form, config) {
+  const inputs = Array.from(form.querySelectorAll(config.inputSelector));
+  const button = form.querySelector(config.submitButtonSelector);
+
+  toggleButtonState(inputs, button);
```

Comment:
**COULD BE IMPROVED**

You can make a `reset` listener for **disabling** the submit button:


```js

  // here you disable the button when you start the project
  toggleButtonState(inputList, buttonElement, config);

  // here you add the `reset` handler
  formElement.addEventListener("reset", () => {
    disableButton(buttonElement, config);
  });

```

In this `reset` listener we are waiting for a special `reset` event which is fired when you **reset** any form